event-sourcing/event-store-jpa/src/main/java/org/apache/james/eventsourcing/eventstore/jpa/model/JPAEvent.java

```diff
@@ -0,0 +1,122 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.eventsourcing.eventstore.jpa.model;
+
+import static org.apache.james.eventsourcing.eventstore.jpa.model.JPAEvent.DELETE_AGGREGATE_QUERY;
+import static org.apache.james.eventsourcing.eventstore.jpa.model.JPAEvent.JPAEventId;
+import static org.apache.james.eventsourcing.eventstore.jpa.model.JPAEvent.SELECT_AGGREGATE_QUERY;
+
+import java.io.Serializable;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.IdClass;
+import jakarta.persistence.Index;
+import jakarta.persistence.NamedQuery;
+import jakarta.persistence.Table;
+
+import org.apache.james.eventsourcing.AggregateId;
+import org.apache.james.eventsourcing.EventId;
+
+import com.google.common.base.Objects;
+
+
+
+/**
+ * JPAEvent class for the James Event Sourcing to be used for JPA persistence.
+ */
+@Entity(name = "JPAEvent")
+@Table(name = JPAEvent.JAMES_EVENTS, indexes = {
+    @Index(name = "AGGREGATE_ID_INDEX", columnList = "AGGREGATE_ID")
+})
+@NamedQuery(name = SELECT_AGGREGATE_QUERY, query = "SELECT e FROM JPAEvent e WHERE e.aggregateId=:aggregateId")
+@NamedQuery(name = DELETE_AGGREGATE_QUERY, query = "DELETE FROM JPAEvent e WHERE e.aggregateId=:aggregateId")
+@IdClass(JPAEventId.class)
+public class JPAEvent {
+    public static final String SELECT_AGGREGATE_QUERY = "selectAggregateEvents";
+    public static final String DELETE_AGGREGATE_QUERY = "deleteAggregateEvents";
+
+    public static final String JAMES_EVENTS = "JAMES_EVENTS";
+
+    public static class JPAEventId implements Serializable {
+
+        private static final long serialVersionUID = 1L;
+
+        private String aggregateId;
+
+        private int eventId;
+
+        public JPAEventId() {
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(aggregateId, eventId);
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null || getClass() != obj.getClass()) {
+                return false;
+            }
+            final JPAEventId other = (JPAEventId) obj;
+            return Objects.equal(this.aggregateId, other.aggregateId)
+                && Objects.equal(this.eventId, other.eventId);
+        }
+    }
+
+    @Id
+    @Column(name = "AGGREGATE_ID", nullable = false, length = 100)
+    private String aggregateId = "";
+
+    @Id
+    @Column(name = "EVENT_ID", nullable = false)
+    private int eventId;
+
+    @Column(name = "EVENT", nullable = false, length = 16 * 1024)
```

Comment:
This may hurt: are we sure this is enough to fit all possible serialized events?

For instance RuleSetDefined may be larger than this for people holding many filters...

Other implementation are unbounded on the size of this field, likely so should do the JPA one.

I suggest looking into using an unbounded `byte[]` here if need be..

Reply:
I have no idea what this is used for and how big the events are, usually events and jsons (used in the other modules as well) are small-ish creatures :-)

I did start with 1mb and it errored, so just picked an arbitrary value to get through the prototype. I suppose if it needs to be a blob or whatnot we can do that. Maybe add to the contract (test) whatever size you think the implementations must support to make sure they're all compatible with the requirements?